Description: Do not bother registering xv/xvmc on gpu screens.
Author: Maarten Lankhorst <maarten.lankhorst@ubuntu.com>
Bug-Ubuntu: https://launchpad.net/bugs/1079096

Index: xorg-server-1.17.2/hw/xfree86/common/xf86xv.c
===================================================================
--- xorg-server-1.17.2.orig/hw/xfree86/common/xf86xv.c
+++ xorg-server-1.17.2/hw/xfree86/common/xf86xv.c
@@ -230,7 +230,7 @@ xf86XVScreenInit(ScreenPtr pScreen, XF86
     ScrnInfoPtr pScrn;
     XF86XVScreenPtr ScreenPriv;
 
-    if (num <= 0 || noXvExtension)
+    if (num <= 0 || noXvExtension || pScreen->isGPU)
         return FALSE;
 
     if (Success != XvScreenInit(pScreen))
Index: xorg-server-1.17.2/hw/xfree86/common/xf86xvmc.c
===================================================================
--- xorg-server-1.17.2.orig/hw/xfree86/common/xf86xvmc.c
+++ xorg-server-1.17.2/hw/xfree86/common/xf86xvmc.c
@@ -148,11 +148,15 @@ xf86XvMCScreenInit(ScreenPtr pScreen,
 {
     XvMCAdaptorPtr pAdapt;
     xf86XvMCScreenPtr pScreenPriv;
-    XvScreenPtr pxvs = (XvScreenPtr) dixLookupPrivate(&pScreen->devPrivates,
-                                                      XF86XvScreenKey);
+    XvScreenPtr pxvs;
     int i, j;
 
-    if (noXvExtension)
+    if (noXvExtension || pScreen->isGPU || !XF86XvScreenKey)
+        return FALSE;
+
+    pxvs = (XvScreenPtr) dixLookupPrivate(&pScreen->devPrivates,
+                                          XF86XvScreenKey);
+    if (!pxvs)
         return FALSE;
 
     if (!(pAdapt = malloc(sizeof(XvMCAdaptorRec) * num_adaptors)))
